Tests/test_numpy.py

```diff
@@ -89,6 +115,24 @@ def test_16bit(self):
         self._test_img_equals_nparray(img, np_img)
         self.assertEqual(np_img.dtype, numpy.dtype('<u2'))
 
+    def test_save_tiff_uint16(self):
+        def _test_image(data, image):
+            '''
+            Check that saving a given image works assuming it's filled with 'data'
+            '''
+            tmpfile = self.tempfile("temp.tif")
+            a = numpy.array([data] * 100, dtype=numpy.uint16)
+            a.shape = TestNumpy.TEST_IMAGE_SIZE
+            Image.fromarray(a).save(tmpfile)
+            im_test = Image.open(tmpfile)
+            im_good = Image.open(image)
+            self.assert_image_equal(im_good, im_test)
+
+        _test_image(0, "Tests/images/uint16_1_0.tif")
+        _test_image(255, "Tests/images/uint16_1_255.tif")
+        _test_image(65280, "Tests/images/uint16_1_65280.tif")
+        _test_image(65535, "Tests/images/uint16_1_65535.tif")
```

Comment:
The problem here is that there are a ton of stages to saving and loading images with Pillow. On loading there's the initial reading of the fileformat, the conversion to an internal array, the conversion to a proper Pillow image class. Now on saving there's the conversion of the internal array to the format specified by the image class and then writing that in an appropriate format to disk.

I've been having a lot of trouble trying to get all of these different pieces in order because there definitely seems to be some lacking support, or incorrect behavior, in various pieces of the chain. This is why I'm interested in adding tests that just improve coverage and should be correct as we'll need a bunch more than this one to test that a uint16 image can be loaded as a uint16 array and then written back out as one.

So are you suggesting that this test be dropped, replaced, or extended with other tests? As I believe the latter is appropriate and what I plan to do once I get this first round of changes done.